docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,222 @@
+
+The [format](#odo-file-format) of this file is described later in the proposal.
+
+### Application related configuration
+
+Some times what we want to configure is how the application/component will be deployed. Some components will
+require special characteristics, like a specific amount of memory. Some other times, what we want is to instruct `odo` to use an specific value to name our component if none is explicitly provided. This configuration will be stored locally to the component, along with the source code, in an `.odo` configuration file that can be saved into the version control system used, and share by anyone using the same application code.
+
+For some specific values, there will be the possibility to set them globally, as they will be used in that case for every component created, as defined in the [configuration precedence](#configuration-precedence) section. This values will be documented as being `global` configuration values.
+
+### Developer friendly flags
+
+One of the goals of `odo` is to be easy to use for developers. This forces to translate some configuration possible in the underlying technology (OpenShift/Kubernetes) into a friendlier name that developers could easily understand. An example of this would be the amount of memory a component needs. In OpenShift, this is reflected by limits.memory and requests.memory, and dependening how you set these values influence the QoS of the deployment. In `odo` these should be simplified to `memory`, `min-memory` and `max-memory`. These are terms that any developer will most likely understand without looking at the documentation.
+
+* When `memory` is set, memory requests and limits are the same.
+* `min-memory` should be memory requests.
+* `max-memory` should be memory limits.
+* `memory` is exclusive with `min-memory` or `max-memory`. If set, min-mem and max-mem have the same value as `memory`.
+
+#### Proposed flags
+
+We need to define and delimit what information will be stored in this file. Configuration which can be local and global will be denoted. Otherwise, the configuration will only be local to the application .odo file.
+
+### Deployment decoration
+
+Configuration used to define how the deployment would be created:
+
+* mem/memory (local/global)
+  * min-mem/min-memory (local/global)
+  * max-mem/max-memory (local/global)
+* cpu (local/global)
+  * min-cpu (local/global)
+  * max-cpu (local/global)
+* health (local)
+  * live-health (local)
+  * ready-health (local)
+* build-image (local)
+  * src-path (local)
+  * artifacts-path (local)
+* runtime-image (local)
+  * src-path (local)
+  * deployment-path (local)
+
+### CLI behavior
+
+Configuration used to define how the CLI should behave:
+
+* app-name: (local) Preferred application name if none is defined
+* component-name: (local) Preferred component name if none is defined
```

Comment:
name or prefix only? and both(app prefix and component prefix) will be same or different?
I assume it can be only prefix right? Because, name seems to suggest that all apps or components will need to be named in entirety as configured...

Reply:
name only. This will be in local .odo config file, it applies only to given component.

Reply:
Ok

Reply:
Fixed